ibis/expr/rules.py

```diff
@@ -432,6 +433,31 @@ def output_dtype(self):
     return output_dtype
 
 
+def _promote_interval_resolution(units: list[IntervalUnit]) -> IntervalUnit:
+    if IntervalUnit.NANOSECOND in units:
+        return IntervalUnit.NANOSECOND
+    elif IntervalUnit.MICROSECOND in units:
+        return IntervalUnit.MICROSECOND
+    elif IntervalUnit.MILLISECOND in units:
+        return IntervalUnit.MILLISECOND
+    elif IntervalUnit.SECOND in units:
+        return IntervalUnit.SECOND
+    elif IntervalUnit.MINUTE in units:
+        return IntervalUnit.MINUTE
+    elif IntervalUnit.HOUR in units:
+        return IntervalUnit.HOUR
+    elif IntervalUnit.DAY in units:
+        return IntervalUnit.DAY
+    elif IntervalUnit.WEEK in units:
+        return IntervalUnit.WEEK
+    elif IntervalUnit.MONTH in units:
+        return IntervalUnit.MONTH
+    elif IntervalUnit.QUARTER in units:
+        return IntervalUnit.QUARTER
+    else:
+        return IntervalUnit.YEAR
```

Comment:
The following might be a bit cleaner:

```suggestion
    # Find the smallest unit present in units
    for unit in reversed(IntervalUnit):
        if unit in units:
            return unit
    assert False, "unreachable"
```

Reply:
Thanks for the suggestion!